File: problem/Count_Hills_and_Valleys/countHillValley.c
int countHillValley(int *nums, int numsSize)
{
    int cnt = 0;
    int j = 0;
    int n = numsSize;

    for (int i = 1; i < n - 1; i++)
    {
        if ((nums[j] < nums[i] && nums[i] > nums[i + 1]) ||
            (nums[j] > nums[i] && nums[i] < nums[i + 1]))
        {
            cnt++;
            j = i;
        }
    }

    return cnt;
}

File: problem/Count_Hills_and_Valleys/countHillValley.js
/**
 * @param {number[]} nums
 * @return {number}
 */
function countHillValley(nums) {
    let cnt = 0;
    let j = 0;
    const n = nums.length;
    
    for (let i = 1; i < n - 1; i++) {
        if ((nums[j] < nums[i] && nums[i] > nums[i + 1]) ||
            (nums[j] > nums[i] && nums[i] < nums[i + 1])) {
            cnt++;
            j = i;
        }
    }
    
    return cnt;
}
